src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs

```
@@ -46,76 +51,108 @@ private enum Direction
+                var responseContent = GetResponseContent(response, requestGuid);
+
+                _agentHealthReporter.ReportSupportabilityDataUsage("Collector", method, uncompressedByteCount, new UTF8Encoding().GetBytes(response.Content.ToString()).Length);
 
                 // Possibly combine these logs? makes parsing harder in tests...
                 Log.DebugFormat("Request({0}): Invoked \"{1}\" with : {2}", requestGuid, method, serializedData);
-                Log.DebugFormat("Request({0}): Invocation of \"{1}\" yielded response : {2}", requestGuid, method, response);
+                Log.DebugFormat("Request({0}): Invocation of \"{1}\" yielded response : {2}", requestGuid, method, responseContent);
 
-                AuditLog(Direction.Received, Source.Collector, response);
+                AuditLog(Direction.Received, Source.Collector, response.Content.ToString());
+                if (!response.IsSuccessStatusCode)
+                {
+                    ThrowExceptionFromHttpResponseMessage(serializedData, response.StatusCode, responseContent, requestGuid);
+                }
 
-                return response;
+                return responseContent;
             }
-            catch (WebException ex)
+            catch (HttpRequestException)
             {
-                var httpWebResponse = ex.Response as HttpWebResponse;
-                if (httpWebResponse != null)
-                    ThrowExceptionFromHttpWebResponse(serializedData, httpWebResponse, requestGuid);
-
                 if (_diagnoseConnectionError)
                     DiagnoseConnectionError(connectionInfo);
-
                 throw;
             }
-        }
-
-        private static string SendRequest(WebRequest request, byte[] requestPayloadData)
-        {
-            SendRequestPayload(request, requestPayloadData);
-            var response = GetResponse(request);
-            return response;
+            catch (Exception ex)
```

In the old code it looks like other kinds of exceptions were ignored. Is there someone above that will catch and/or log this?